platform/viewer/src/routes/DataSourceWrapper.jsx

```diff
@@ -49,6 +49,7 @@ function DataSourceWrapper(props) {
     total: 0,
     resultsPerPage: 25,
     pageNumber: 1,
+    location: 'PERFORM_INITIAL_QUERY',
```

Comment:
The changes in this file prevent endlessly querying the studies when no results are found, and cause the actual query to be done whenever the query parameters change.  That means the query is done correctly initially, as the currently querying results are specified in "location".  There should be a query update prevention when there is an outstanding query, and the parameters change before it comes back.

Reply:
I need more explanation for changes on this file to review it. I'll send a message

Reply:
Provided offline